src/utils/bintools.ts

```diff
@@ -284,12 +285,57 @@ export default class BinTools {
   addressToString = (hrp:string, chainid:string, bytes:Buffer)
   :string => `${chainid}-${bech32.encode(hrp, bech32.toWords(bytes))}`;
 
-  stringToAddress = (address:string):Buffer => {
-    const parts:Array<string> = address.trim().split('-');
-    if(parts[1].startsWith("0x") || parts[1].match(/^[0-9A-F]+$/i)){
-      return Buffer.from(parts[1].replace("0x", ""), "hex");
+  stringToAddress = (address: string, hrp?: string): Buffer => {
+    const validteETHStyleAddress = (a: string) => {
+      if(utils.isAddress(a)) {
+        return Buffer.from(a.replace("0x", ""), "hex");
+      } else {
+        throw new HexError('Error - Invalid address');
+      }
+    };
+    if(address.substring(0, 2) === "0x") {
+      // ETH-style address
+      try {
+        return validteETHStyleAddress(address);
+      } catch (error) {
+        throw error;
+      }
+    } else {
+      // Bech32 addresses
+      const parts: string[] = address.trim().split('-');
+
+      if(parts.length < 2) {
+        throw new Bech32Error('Error - Valid address should include -');
+      }
+
+      if(parts[0].length < 1) {
+        throw new Bech32Error('Error - Valid address must have prefix before -');
+      }
+
+      if(parts[1].substring(0, 2) === "0x") {
```

Comment:
I don't think we need this because we check above whether it's a hex address, and I don't think we consider `[prefix]-0x...` to be a valid address

Reply:
Fixed